deeprankcore/Trainer.py

```diff
@@ -582,49 +592,34 @@ def log_epoch_data(stage, epoch, loss, time):
             'Epoch [%04d] : %s loss %e | time %1.2e sec.' % # pylint: disable=consider-using-f-string
             (epoch, stage, loss, time))
 
-    def format_output(self, pred, target=None):
+    def _format_output(self, pred, target=None):
         """Format the network output depending on the task (classification/regression)."""
-        if self.task == "class":
-            # pred = F.softmax(pred, dim=1)
-
-            if target is not None:
-                target = torch.tensor([self.classes_to_idx[int(x)]
-                                       for x in target]).to(self.device)
 
-        elif self.transform_sigmoid is True:
+        if (self.task == 'class') and (target is not None):
+            # For categorical cross entropy, the target must be a one-dimensional tensor
+            # of class indices with type long and the output should have raw, unnormalized values
+            classes_to_idx = {
+                i: idx for idx, i in enumerate(self.classes)
+            }
 
-            # Sigmoid(x) = 1 / (1 + exp(-x))
-            pred = torch.sigmoid(pred.reshape(-1))
+            target = torch.tensor(
+                [classes_to_idx[int(x)] for x in target]
```

Comment:
I know that the classes_to_idx translation table isn't used much, since we always work with numerical targets anyway. However, I do think is should be a property of the run, rather than a temporary value. After all, this information is needed at the end of the run to convert the indexes back into class labels.

Reply:
You're right, I moved it back to `_put_model_to_device` and I made it an attribute of Trainer again.